### PR TITLE
First round of ClipScrollTree optiizations

### PR DESCRIPTION
**Pull request by mrobinson**
These optimizations reduce the prominence of `ClipScrollTree::update_tree` in my perf CPU profile.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/webrender/2142)

### CODE REVIEW

**Comment by emilio** [comment]
(You could use `Rc<RefCell<..>>`, though, I guess... Though it's kinda ugly)

**Comment by mrobinson** [comment]
@emilio: Thanks. I'll just omit that commit for now and try a reference counted type tomorrow. Not sure what I was thinking!

**Comment by glennw** [comment]
This looks good to me. We should do a gecko try run first, then r=me if all green.

**Comment by mrobinson** [comment]
The try job is sitting at 99% with only localization test yet to finish (it's been running all day). There is one failure, but it is a JS test timeout (known intermittent). I think it's okay to hand this one to bors.

https://treeherder.mozilla.org/#/jobs?repo=try&revision=672c6645e527433f72a0939ce4f981cc93ba56ef&group_state=expanded

File: webrender/res/prim_shared.glsl [hunk 1]
```diff
@@ -548,9 +539,11 @@ vec4 get_layer_pos(vec2 pos, Layer layer) {
     // get a point on the layer plane
     vec4 ah = layer.transform * vec4(0.0, 0.0, 0.0, 1.0);
     vec3 a = ah.xyz / ah.w;
+
     // get the normal to the layer plane
-    vec3 n = transpose(mat3(layer.inv_transform)) * vec3(0.0, 0.0, 1.0);
-    return untransform(pos, n, a, layer.inv_transform);
+    mat4 inv_transform = inverse(layer.transform);
```
Comment (kvark) [inline webrender/res/prim_shared.glsl]: I don't think it's a good idea to compute the inverse mat4 per vertex.

Comment (mrobinson) [reply webrender/res/prim_shared.glsl]: So I definitely want to do some more testing here, but I believe this is only called for clips. In any case, it's probably pretty easy to move this back to the CPU only for nodes that need it.

**Pull request merged**